tests/pos/i4322.scala

```diff
@@ -0,0 +1,10 @@
+object Foo {
+  private[this] var x: Int = 1
+
+  inline def foo: Int = x + x + x
```

Comment:
I suggest turning this into a run test and then checking that there's only one accessor method using `Class#getMethods` (see https://github.com/lampepfl/dotty/blob/master/tests/run/traitNoInit.scala for an example on how to do something like that)